### PR TITLE
Add validation for als, bpr, and lmf

### PR DESCRIPTION
**Pull request by ita9naiwa**
add validation via log.info.

### CODE REVIEW

**Comment by benfred** [comment]
I think there is a better way of doing this, and that we should revert these changes =(.

The models shouldn't be responsible for validation: the eval code should be capable of validating any model, and we shouldn't have to add evaluation code to the models themselves. 

Was this code added to solve https://github.com/benfred/implicit/issues/281 ? 

We already have a solution for calculating/plotting training loss at each iteration. As an example, for [this post](http://www.benfrederickson.com/fast-implicit-matrix-factorization/) I plotted out the training loss for cg/cholesky at each iteration:

CG Loss

This was done with the 'fit_callback' code - which lets you hook in any arbitrary function after each iteration. This can also be used to do cross-validation after each iteration - and could be easily added to both the BPR/LMF models.  The code to collect these models is here https://github.com/benfred/implicit/blob/master/benchmarks/benchmark_als.py#L25-L54 .  

Alternatively, we could also go with a partial fit approach (so you can call fit multiple times and validate in between) - but I don't think that the models themselves should be responsible for running validation like done here.  The idea behind the 'fit' call is to (very) roughly mimic interface from sklearn - and this significantly diverges from that.

**Comment by ita9naiwa** [comment]
Adding valiadation is not a good idea.

**Comment by ita9naiwa** [comment]
@benfred 
All commits from this were reverted.

Could you please remove version 0.4.2 of implicit from pypi?..
